### PR TITLE
Add support for postgres schema selection

### PR DESCRIPTION
**Pull request by sirudog**
Currently the schema is hardcoded to `'public'`.
This PR adds support for postgres schema selection by using the options parameter in the connection string, which is the proper Postgres-native way. 

The connection string would look like (before URL encoding):
`postgresql://user:pass@host:port/database?options=-c search_path=custom_schema`

How it works:

  - Native Postgres handling - `psycopg2/libpq` processes the options parameter directly, so `search_path` is set automatically on connection
  - Parse for metadata queries - Extract `search_path` value from the options parameter to use in `information_schema` queries
  - Default to public - If no options or no `search_path` specified, fall back to `'public'`

Why this approach:
  - Minimal changes - Only postgres_loader.py needs modification
  - Backward compatible - Defaults to `'public'` if not specified
  - User-friendly - Follows familiar URL parameter pattern
  - Postgres-specific - Doesn't affect MySQL or other loaders

## Summary by CodeRabbit

* **New Features**
  * Added support for specifying PostgreSQL schemas via connection URLs and a new schema input field in the database connection modal.
  * Users can now target specific schemas instead of defaulting to "public."

* **Documentation**
  * Updated PostgreSQL loader documentation with schema configuration guide, examples, and troubleshooting steps.

* **Tests**
  * Added comprehensive test suite for schema parsing functionality covering edge cases and various input formats.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by galshubeli** [comment]
@sirudog Thanks for the contribution

File: app/src/components/modals/DatabaseModal.tsx [hunk 1]
```diff
@@ -94,6 +95,12 @@ const DatabaseModal = ({ open, onOpenChange }: DatabaseModalProps) => {
         builtUrl.username = username;
         builtUrl.password = password;
         dbUrl = builtUrl.toString();
+
+        // Append schema option for PostgreSQL if provided
+        if (selectedDatabase === 'postgresql' && schema.trim()) {
+          const schemaOption = `options=-csearch_path%3D${encodeURIComponent(schema.trim())}`;
+          dbUrl += (dbUrl.includes('?') ? '&' : '?') + schemaOption;
```
Comment (galshubeli) [inline app/src/components/modals/DatabaseModal.tsx]: better to use:
          const url = new URL(dbUrl);
          url.searchParams.set('options', `-csearch_path=${schema.trim()}`);
          dbUrl = url.toString();

**Comment by galshubeli** [comment]
@sirudog I left single comment. Please fix the spellcheck

File: app/src/components/modals/DatabaseModal.tsx (hunk 1 above)
Comment (sirudog) [reply app/src/components/modals/DatabaseModal.tsx]: Indeed, fixed

**Comment by sirudog** [comment]
@galshubeli  I'll revisit the spell checks... There are still a few :-(

**Comment by sirudog** [comment]
> @galshubeli I'll revisit the spell checks... There are still a few :-(

@galshubeli, it is done, I hope I did not miss any this time :-)

**Review by gkorland — APPROVED** [review]
